src/freenas/etc/ix.rc.d/ix-ldap

```diff
@@ -134,13 +134,34 @@ ldap_status()
 		local res=1
 		local ldapsearch=/usr/local/bin/ldapsearch
 		local options=
-
-		if [ "${ldap_ssl}" = "start_tls" ]
+		local do_gssapi_bind=0
+		local default_ssl="off"
+		local force_gssapi="${FREENAS_LDAP_FORCE_GSSAPI:-1}"
+
+		# Do GSSAPI bind if SSL and Anonymous binding or
+		# disabled and we have a keytab + realm or
+		# if we have a keytab+realm and FORCE_GSSAPI is set.
+		if [ "${ldap_ssl:-$default_ssl}" = "start_tls" ]
 		then
+			do_gssapi_bind=1
 			options="-Z"
+
+		elif [ "${ldap_ssl:-$default_ssl}" = "on" ]
+		then
+			do_gssapi_bind=1
+		fi
+
+		if [ "${ldap_anonbind}" = "1" ]
+		then
+			do_gssapi_bind=1
+		fi
+
+		if [ "${force_gssapi}" = "0" ]
+		then
+			do_gssapi_bind=0
 		fi
 
-		if [ -n "${realm}" -a -n "${principal}" ]
+		if [ -n "${realm}" -a -n "${principal}" -a "${do_gssapi_bind}" == 0 ]
```

Comment:
Hmm, I don't think `==` works in shell, i think you're looking for `${do_gssapi_bind} -eq 0` ?